Sources/StreamChat/Database/DTOs/AttachmentDTO.swift

```diff
@@ -150,68 +143,64 @@ extension NSManagedObjectContext: AttachmentDatabaseSession {
 
 extension AttachmentDTO {
     /// Snapshots the current state of `AttachmentDTO` and returns an immutable model object from it.
-    func asModel<ExtraData: ExtraDataTypes>() -> _ChatMessageAttachment<ExtraData> { .create(fromDTO: self) }
+    func asModel() -> DatabaseAttachment {
+        let type = AttachmentType(rawValue: self.type)
+        
+        switch type {
+        case .custom:
+            return ChatMessageCustomAttachment(id: attachmentID, type: type, data: data)
+        default:
+            guard
+                let data = data,
+                var defaultAttachment = try? JSONDecoder.default.decode(ChatMessageDefaultAttachment.self, from: data)
+            else {
+                log.error(
+                    "Unable to decode `ChatMessageDefaultAttachment` for built-in type." +
+                        "Falling back to ChatMessageCustomAttachment"
+                )
+                return ChatMessageCustomAttachment(id: attachmentID, type: type, data: self.data)
+            }
+            defaultAttachment.id = attachmentID
+            defaultAttachment.localURL = localURL
+            defaultAttachment.localState = localState
+            return defaultAttachment
+        }
+    }
     
-    /// Snapshots the current state of `AttachmentDTO` and returns its representation for used in API calls.
-    func asRequestPayload<ExtraData: AttachmentExtraData>() -> AttachmentRequestBody<ExtraData> { .create(fromDTO: self) }
-}
-
-private extension _ChatMessageAttachment {
-    /// Create a ChatMessageAttachment  struct from its DTO
-    static func create(fromDTO dto: AttachmentDTO) -> _ChatMessageAttachment {
-        let extraData: ExtraData.Attachment
-        do {
-            extraData = try JSONDecoder.default.decode(ExtraData.Attachment.self, from: dto.extraData)
-        } catch {
-            log.error(
-                "Failed to decode extra data for Attachment with hash: <\(dto.attachmentID)>, using default value instead. "
-                    + "Error: \(error)"
-            )
-            extraData = .defaultValue
+    /// Returns an object pending to upload.
+    func asUploadingModel() -> ChatMessageDefaultAttachment? {
```

Comment:
That is why I was saying that `Probably not uploaded attachments should be another type.`
This helper is used in `AttachmentUploader` and it ensures that the attachment has local data needed for uploading and avoid extra casting from `DatabaseAttachment` to `ChatMessageDefaultAttachment`.
It can be removed but in this case `asModel` helper will become more complicated.